### PR TITLE
Dashboard image management

### PR DESCRIPTION
**Pull request by artursmet**
* Allow to pick variants for images

image

### CODE REVIEW

File: saleor/dashboard/product/forms.py [hunk 1]
```diff
@@ -138,9 +142,28 @@ class Meta:
...
+    def save_variant_images(self, instance):
+        variant_images = []
+        # Clean up old mapping
+        instance.variant_images.all().delete()
+        for variant in self.cleaned_data['variants']:
+            variant_images.append(
+                VariantImage(variant=variant, image=instance))
+        VariantImage.objects.bulk_create(variant_images)
+
+    def save(self, commit=True):
```
Comment (mociepka) [inline saleor/dashboard/product/forms.py]: Atomic transaction will be nice here.

File: saleor/product/forms.py [hunk 2]
```diff
@@ -34,8 +37,14 @@ class ProductForm(AddToCartForm):
 
     def __init__(self, *args, **kwargs):
         super(ProductForm, self).__init__(*args, **kwargs)
-        self.fields['variant'].queryset = self.product.variants
-        self.fields['variant'].empty_label = None
+        variant = self.fields['variant']
```
Comment (mociepka) [inline saleor/product/forms.py]: Can we name it `variant_field`?

File: saleor/dashboard/product/forms.py [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 
 from django import forms
+from django.db import transaction
 from django.forms.models import inlineformset_factory, ModelChoiceIterator
 from django.utils.translation import pgettext_lazy
 
 from ...product.models import (ProductImage, Stock, ProductVariant, Product,
-                               ProductAttribute, AttributeChoiceValue)
+                               ProductAttribute, AttributeChoiceValue,
+                               VariantImage)
```
Comment (patrys) [inline saleor/dashboard/product/forms.py]: Please run `isort` on this file.

File: saleor/dashboard/product/forms.py [hunk 4]
```diff
@@ -138,9 +143,29 @@ class Meta:
...
+            variants.queryset = self.instance.product.variants.all()
+            variants.initial = self.instance.variant_images.values_list(
+                'variant', flat=True)
         if self.instance.image:
             self.fields['image'].widget = ImagePreviewWidget()
 
+    def save_variant_images(self, instance):
+        variant_images = []
+        # Clean up old mapping
+        with transaction.atomic():
```
Comment (patrys) [inline saleor/dashboard/product/forms.py]: Since this function only does one thing, you can decorate it instead of using the context manager.

**Comment by artursmet** [comment]
@patrys I sorted imports and used atomic as decorator instead of context manager

**Comment by patrys** [comment]
LGTM. Any takers for the JS counterpart? Otherwise merging will result in confusion.

**Comment by patrys** [comment]
@car3oon If you find a minute, the remaining work here is to automatically switch the carousel (product details page) to the appropriate image once a variant is picked.

**Pull request merged**